_posts/en/newsletters/2022-09-14-newsletter.md

```diff
@@ -0,0 +1,176 @@
+  `m_headers_sync_timeout`, and `PeerManagerImpl::nSyncStarted`?
+  If we start syncing headers with peers that announce a block to
+  us via an `inv`, why do we not increase `nSyncStarted` and set
+  `fSyncStarted = true` and update `m_headers_sync_timeout`?"
+  a3="`nSyncStarted` counts the number of peers whose `fSyncStarted`
+  is true, and this number can't be greater than 1 until the
+  node has headers close to (within one day of) the current time.
+  This (arbitrary) peer will be our initial headers-sync peer. If this
+  peer is slow, the node times it out (`m_headers_sync_timeout`) and
+  finds another 'initial' headers syncing peer.
+  But if, during headers sync, a node sends an `inv` message
+  that announces blocks, then without this PR, the node will
+  start requesting headers from this peer as well, _without_ setting
+  its `fSyncStarted` flag. This is the source of the redundant
+  headers messages, and was probably not intended, but has the
+  benefit of allowing headers sync to proceed even if the
+  initial headers-sync peer is malicious, broken, or very slow.
+  With this PR, the node requests headers from only _one_ additional
+  peer (rather than from all peers who announced the new block)."
+  a3link="https://bitcoincore.reviews/25720#l-102"
+
+  q4="An alternative to the approach taken in the PR would be to add
+  an additional headers sync peer after a timeout (fixed or random).
+  What is the benefit of the approach taken in the PR over this
+  alternative?"
+  a4="One benefit is that peers that announce an `inv` to us have a
+  higher probability of being responsive. Another is that a peer that
+  manages to send us the block `inv` first is often also a very fast peer.
+  So we'd not pick another slow peer if for some reason our initial
+  peer is slow."
+  q4link="https://bitcoincore.reviews/25720#l-135"
+%}
+
+## Releases and release candidates
+
+*New releases and release candidates for popular Bitcoin infrastructure
+projects.  Please consider upgrading to new releases or helping to test
+release candidates.*
+
+- [LDK 0.0.111][] adds support for creating, receiving, and relaying
+  [onion messages][topic onion messages], among several other new
+  features and bug fixes.
+
+## Notable code and documentation changes
+
+*Notable changes this week in [Bitcoin Core][bitcoin core repo], [Core
+Lightning][core lightning repo], [Eclair][eclair repo], [LDK][ldk repo],
+[LND][lnd repo], [libsecp256k1][libsecp256k1 repo], [Hardware Wallet
+Interface (HWI)][hwi repo], [Rust Bitcoin][rust bitcoin repo], [BTCPay
+Server][btcpay server repo], [BDK][bdk repo], [Bitcoin Improvement
+Proposals (BIPs)][bips repo], and [Lightning BOLTs][bolts repo].*
+
+- [Bitcoin Core #25614][] builds on [Bitcoin Core #24464][], allowing
+  the ability to add and trace logs with specific severity levels in
+  addrdb, addrman, banman, i2p, mempool, netbase, net, net_processing,
+  timedata, and torcontrol.
```

Comment:
Thanks for covering this change. It added a `-loglevel` configuration option to enable users to select global and per-category severity log levels as part of a larger, ongoing migration to logging with severity levels. It is a "hidden" debug option for now.